data/writing-tests.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @title Writing Tests
+ * @difficulty beginner
+ * @tags cli
+ * @run deno test --allow-read --allow-write <url>
+ * @resource {https://deno.land/api?s=Deno.test} Doc: Deno.test
+ * @resource {$std/testing/asserts.ts} Doc: std/testing/asserts
+ *
+ * One of the most common tasks in developing software is writing tests for
+ * existing code. Deno has a built-in test runner which makes this very easy.
+ */
+
+// First, we import assert statements from the standard library. There are
+// quite a few options but we will just import the most common ones here.
+import { assert, assertEquals } from "$std/testing/asserts.ts";
+
+// The most simple way to use the test runner is to just pass it a description
+// and a callback function
+Deno.test("assert works correctly", () => {
+  assert(true);
+  assertEquals(1, 1);
+});
+
+// In more complex scenarios, we often need to have some setup and teardown code
+// with some steps in between. This is also made simple with the built-in test runner.
+Deno.test("testing steps", async (t) => {
+  const file = await Deno.open("example.txt", {
+    read: true,
+    write: true,
+    create: true,
+  });
+  const encoder = new TextEncoder();
+  const data = encoder.encode("Hello world!");
+
+  await t.step("write some bytes", async () => {
+    await file.write(data);
```

Comment:
Maybe let's assert about the written byte length? I think the test case without any assertion might look confusing to beginners.

Reply:
Makes sense, let's do that.